lib/dor/services/client/marcxml.rb

```diff
@@ -21,6 +23,37 @@ def catkey(barcode:)
 
           raise_exception_based_on_response!(resp)
         end
+
+        # Gets MARCXML corresponding to a barcode or catkey
+        # @param barcode [String] required string representing a barcode
+        # @param catkey [String] required string representing a catkey
+        # @raise [NotFoundResponse] when the response is a 500 with "Record not found in Symphony"
+        # @raise [UnexpectedResponse] on an unsuccessful response from the server
+        # @return [Nokogiri::XML::Document] MARCXML
+        # rubocop:disable Metrics/AbcSize
+        def marcxml(barcode: nil, catkey: nil)
+          check_args(barcode, catkey)
+
+          resp = connection.get do |req|
+            req.url "#{api_version}/catalog/marcxml"
+            req.params['barcode'] = barcode unless barcode.nil?
+            req.params['catkey'] = catkey unless catkey.nil?
+          end
+
+          raise NotFoundResponse if !resp.success? && resp.body.match?(/Record not found in Symphony/)
```

Comment:
I wonder if this class should customize `raise_exception_based_on_response` since it has different requirements than the other classes. And then this logic can move into that method.

Reply:
Done.